hdfs-dataset-sink/src/main/java/org/springframework/cloud/stream/module/dataset/sink/DatasetSinkConfiguration.java

```diff
@@ -0,0 +1,190 @@
+		aggregatorFactoryBean.setMessageStore(messageGroupStore);
+		aggregatorFactoryBean.setProcessorBean(new DefaultAggregatingMessageGroupProcessor());
+		aggregatorFactoryBean.setExpireGroupsUponCompletion(true);
+		aggregatorFactoryBean.setSendPartialResultOnExpiry(true);
+		aggregatorFactoryBean.setOutputChannel(toSink);
+		return aggregatorFactoryBean;
+	}
+
+	@Bean
+	MessageGroupStore messageGroupStore() {
+		SimpleMessageStore messageGroupStore = new SimpleMessageStore();
+		messageGroupStore.setTimeoutOnIdle(true);
+		messageGroupStore.setCopyOnGet(false);
+		return messageGroupStore;
+	}
+
+	@Bean
+	MessageGroupStoreReaper messageGroupStoreReaper(MessageGroupStore messageStore) {
+		MessageGroupStoreReaper messageGroupStoreReaper = new MessageGroupStoreReaper(messageStore);
+			messageGroupStoreReaper.setTimeout(properties.getIdleTimeout());
+		messageGroupStoreReaper.setAutoStartup(true);
+		messageGroupStoreReaper.setExpireOnDestroy(true);
+		return messageGroupStoreReaper;
+	}
+
+	@Bean
+	Reaper reaper() {
+		return new Reaper();
+	}
+
+	@Bean
+	public DatasetOperations datasetOperations(DatasetRepositoryFactory datasetRepositoryFactory,
+	                                           DatasetDefinition datasetDefinition) {
+		return new DatasetTemplate(datasetRepositoryFactory, datasetDefinition);
+	}
+
+	@Bean
+	public DatasetRepositoryFactory datasetRepositoryFactory(org.apache.hadoop.conf.Configuration configuration) {
+		DatasetRepositoryFactory datasetRepositoryFactory = new DatasetRepositoryFactory();
+		datasetRepositoryFactory.setConf(configuration);
+		datasetRepositoryFactory.setBasePath(properties.getDirectory());
+		datasetRepositoryFactory.setNamespace(properties.getNamespace());
+		return datasetRepositoryFactory;
+	}
+
+	@Bean
+	public DatasetDefinition datasetDefinition() {
+
+		DatasetDefinition datasetDefinition =
+				new DatasetDefinition(properties.isAllowNullValues(), properties.getFormat());
+		if (StringUtils.hasText(properties.getPartitionPath())) {
+			datasetDefinition.setPartitionStrategy(parsePartitionExpression(properties.getPartitionPath()));
+		}
+		if (properties.getWriterCacheSize() > 0) {
+			datasetDefinition.setWriterCacheSize(properties.getWriterCacheSize());
+		}
+		if (StringUtils.hasText(properties.getCompressionType())) {
+			datasetDefinition.setCompressionType(properties.getCompressionType());
+		}
+		return datasetDefinition;
+	}
+
+	private static PartitionStrategy parsePartitionExpression(String expression) {
+
+		List<String> expressions = Arrays.asList(expression.split("/"));
+
+		ExpressionParser parser = new SpelExpressionParser();
+		PartitionStrategy.Builder psb = new PartitionStrategy.Builder();
+		StandardEvaluationContext ctx = new StandardEvaluationContext(psb);
+		for (String expr : expressions) {
+			try {
+				Expression e = parser.parseExpression(expr);
+				psb = e.getValue(ctx, PartitionStrategy.Builder.class);
+			}
+			catch (SpelParseException spe) {
```

Comment:
trying to keep the functionality from XD where partition expressions are separated by '/' character